t/01_opensearch_modules.t

```diff
@@ -3,7 +3,7 @@ use Test::More 0.98;
 use Data::Dumper;
 
 plan skip_all => 'To test Modules, set OS_HOST, OS_USER, OS_PASS, OS_INDEX in ENV'
-  unless $ENV{OS_HOST} && $ENV{OS_USER} && $ENV{OS_PASS} && $ENV{OS_INDEX};
+  unless $ENV{OS_HOST} && exists($ENV{OS_USER}) && exists($ENV{OS_PASS}) && $ENV{OS_INDEX};
```

Comment:
I made some changes here to make the tests run:

* `OS_USER` and `OS_PASS` are optional
* connections are not secure (it tried to connect over https). If it is needed, additional environments variables should be made

Reply:
At least in Ubuntu when using the Opensearch repository (https://artifacts.opensearch.org/releases/bundle/opensearch/2.x/apt)
ssl defaults to true: `plugins.security.ssl.http.enabled: true`
Thats probably why i set it as default.

Thank you for all the work you've done so far. Il try it out and merge when my testing is done.